src/source/internal.rs

```diff
@@ -316,10 +316,10 @@ impl Source for Internal {
                     }
                 }
             } else {
-                // do nothing, intentionally
+                while let Some(_) = Q.pop() {
```

Comment:
Is there a chance we block here?

Reply:
There isn't. The internal implementation of `coco::Stack` might take more or less time to return an item from the stack but when an item is not available `None` will be returned and we'll exit the loop. 

Now, that said, there is the possibility that either of these loops -- in this branch or the one, similar, above -- will be saturated by points and will be unable to exit. I was unable to trigger this condition even with a custom-built source whose sole role was spamming the Internal stack. I believe this is a low-probability condition and, even so, is something that's inline with "process all the points available".